Maven tries to load `pom.yaml` as XML

I have a simple multi-project setup (https://github.com/ueberfuhr-playgrounds/maven4/tree/polyglot-pom) where I try to use Mason for YAML syntax. I have added the extension to `.mvn/extensions.xml`, and the extension is downloaded, so recognized by Maven. But Maven seems to try to read the `pom.yaml` still as XML, as I get the error:

```bash
Non-parseable POM /maven4-samples/pom.yaml: Unable to read model: Unexpected character 'm' (code 109) in prolog; expected '<'
```

What am I missing here? The GitHub action is running with Maven 4 RC 5.

Just FYI, w/ latest changes on main, I still have this locally when modify project-a and project-b POMs (making them empty):
```
[cstamas@angeleyes maven4 (polyglot-pom *)]$ git diff
diff --git a/.mvn/extensions.xml b/.mvn/extensions.xml
index 803a68d..1ad9fb8 100644
--- a/.mvn/extensions.xml
+++ b/.mvn/extensions.xml
@@ -3,6 +3,6 @@
   <extension>
     <groupId>eu.maveniverse.maven.mason</groupId>
     <artifactId>mason</artifactId>
-    <version>0.2.1</version>
+    <version>0.2.2-SNAPSHOT</version>
   </extension>
 </extensions>
diff --git a/project-a/pom.yaml b/project-a/pom.yaml
index 0f9269e..861ec01 100644
--- a/project-a/pom.yaml
+++ b/project-a/pom.yaml
@@ -1,6 +1,6 @@
 modelVersion: 4.1.0
 # should be empty (https://github.com/maveniverse/mason/pull/36)
-parent: com.samples:maven4-samples:1.0-SNAPSHOT
+parent:
 id: com.samples.maven4-samples:project-a
 packaging: jar
 dependencies:
diff --git a/project-b/pom.yaml b/project-b/pom.yaml
index da4ee67..dccd8ca 100644
--- a/project-b/pom.yaml
+++ b/project-b/pom.yaml
@@ -1,6 +1,6 @@
 modelVersion: 4.1.0
 # should be empty (https://github.com/maveniverse/mason/pull/36)
-parent: com.samples:maven4-samples:1.0-SNAPSHOT
+parent:
 id: com.samples.maven4-samples:project-b
 packaging: jar
 dependencies:
[cstamas@angeleyes maven4 (polyglot-pom *)]$ mvn -V validate -e
Apache Maven 4.0.0-rc-5 (fb3ecaef88106acb40467a450248dfdbd75f3b35)
Maven home: /home/cstamas/.sdkman/candidates/maven/4.0.0-rc-5
Java version: 21.0.8, vendor: Eclipse Adoptium, runtime: /home/cstamas/.sdkman/candidates/java/21.0.8-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.17.10-300.fc43.x86_64", arch: "amd64", family: "unix"
[INFO] Loaded 21762 auto-discovered prefixes for remote repository central (prefixes-central.txt)
[INFO] Error stacktraces are turned on.
[INFO] 
[INFO] 1 problem was encountered while building the effective settings (use -e to see details)
[WARNING] Detected 1 pre-Maven 4 legacy encrypted password(s) - configure password encryption with the help of mvnenc for increased security. @ /home/cstamas/.m2/settings.xml
[INFO] 
[INFO] Scanning for projects...
[ERROR] Some problems were encountered while processing the POMs
[ERROR] The build could not read 1 project -> [Help 1]
org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs
    at org.apache.maven.project.DefaultProjectBuilder$BuildSession.build(DefaultProjectBuilder.java:511)
    at org.apache.maven.project.DefaultProjectBuilder.build(DefaultProjectBuilder.java:194)
    at org.apache.maven.project.collector.DefaultProjectsSelector.selectProjects(DefaultProjectsSelector.java:61)
    at org.apache.maven.project.collector.RequestPomCollectionStrategy.collectProjects(RequestPomCollectionStrategy.java:49)
    at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor(DefaultGraphBuilder.java:367)
    at org.apache.maven.graph.DefaultGraphBuilder.build(DefaultGraphBuilder.java:100)
    at org.apache.maven.DefaultMaven.buildGraph(DefaultMaven.java:643)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:250)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:225)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:149)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.doExecute(MavenInvoker.java:452)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:97)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:81)
    at org.apache.maven.cling.invoker.LookupInvoker.doInvoke(LookupInvoker.java:165)
    at org.apache.maven.cling.invoker.LookupInvoker.invoke(LookupInvoker.java:134)
    at org.apache.maven.cling.ClingSupport.run(ClingSupport.java:76)
    at org.apache.maven.cling.MavenCling.main(MavenCling.java:51)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke(Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
[ERROR]   
[ERROR]   The project com.samples:maven4-samples:pom:1.0-SNAPSHOT (/home/cstamas/tmp/ueberfur/maven4/pom.yaml) has 2 errors
[ERROR]     Failed to load project /home/cstamas/tmp/ueberfur/maven4/project-b/pom.yaml @ com.samples:maven4-samples:1.0-SNAPSHOT, /home/cstamas/tmp/ueberfur/maven4/pom.yaml: 1 problem was 
[ERROR]         - [FATAL] Non-parseable POM /home/cstamas/tmp/ueberfur/maven4/project-b/pom.yaml: Unable to read model: Unexpected character 'm' (code 109) in prolog; expected '<'
[ERROR]      at [row,col,system-id]: [1,1,"file:///home/cstamas/tmp/ueberfur/maven4/project-b/pom.yaml"]
org.apache.maven.api.services.ModelBuilderException: 1 problem was 
    - [FATAL] Non-parseable POM /home/cstamas/tmp/ueberfur/maven4/project-b/pom.yaml: Unable to read model: Unexpected character 'm' (code 109) in prolog; expected '<'
 at [row,col,system-id]: [1,1,"file:///home/cstamas/tmp/ueberfur/maven4/project-b/pom.yaml"]
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.newModelBuilderException(DefaultModelBuilder.java:523)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.doReadFileModel(DefaultModelBuilder.java:1508)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.lambda$cache$14(DefaultModelBuilder.java:2094)
    at org.apache.maven.impl.cache.CachingSupplier.apply(CachingSupplier.java:50)
    at org.apache.maven.impl.cache.AbstractRequestCache.request(AbstractRequestCache.java:63)
    at org.apache.maven.impl.AbstractSession.request(AbstractSession.java:171)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.cache(DefaultModelBuilder.java:2094)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.readFileModel(DefaultModelBuilder.java:1433)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.loadFilePom(DefaultModelBuilder.java:851)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.lambda$loadFilePom$7(DefaultModelBuilder.java:912)
    at org.apache.maven.impl.util.PhasingExecutor.lambda$execute$0(PhasingExecutor.java:80)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.lang.Thread.run(Thread.java:1583)
[ERROR]     Failed to load project /home/cstamas/tmp/ueberfur/maven4/project-a/pom.yaml @ com.samples:maven4-samples:1.0-SNAPSHOT, /home/cstamas/tmp/ueberfur/maven4/pom.yaml: 1 problem was 
[ERROR]         - [FATAL] Non-parseable POM /home/cstamas/tmp/ueberfur/maven4/project-a/pom.yaml: Unable to read model: Unexpected character 'm' (code 109) in prolog; expected '<'
[ERROR]      at [row,col,system-id]: [1,1,"file:///home/cstamas/tmp/ueberfur/maven4/project-a/pom.yaml"]
org.apache.maven.api.services.ModelBuilderException: 1 problem was 
    - [FATAL] Non-parseable POM /home/cstamas/tmp/ueberfur/maven4/project-a/pom.yaml: Unable to read model: Unexpected character 'm' (code 109) in prolog; expected '<'
 at [row,col,system-id]: [1,1,"file:///home/cstamas/tmp/ueberfur/maven4/project-a/pom.yaml"]
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.newModelBuilderException(DefaultModelBuilder.java:523)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.doReadFileModel(DefaultModelBuilder.java:1508)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.lambda$cache$14(DefaultModelBuilder.java:2094)
    at org.apache.maven.impl.cache.CachingSupplier.apply(CachingSupplier.java:50)
    at org.apache.maven.impl.cache.AbstractRequestCache.request(AbstractRequestCache.java:63)
    at org.apache.maven.impl.AbstractSession.request(AbstractSession.java:171)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.cache(DefaultModelBuilder.java:2094)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.readFileModel(DefaultModelBuilder.java:1433)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.loadFilePom(DefaultModelBuilder.java:851)
    at org.apache.maven.impl.model.DefaultModelBuilder$ModelBuilderSessionState.lambda$loadFilePom$7(DefaultModelBuilder.java:912)
    at org.apache.maven.impl.util.PhasingExecutor.lambda$execute$0(PhasingExecutor.java:80)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.lang.Thread.run(Thread.java:1583)
[ERROR] 
[ERROR] Re-run Maven using the '-X' switch to enable verbose output
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[cstamas@angeleyes maven4 (polyglot-pom *)]$ 
```

--- a/README.md
+++ b/README.md
@@ -13,19 +13,32 @@ Mason is a Maven extension that provides alternative model readers for Maven POM
 * Java 17+
 * Maven 4.0.0-rc-3+
 
-## Usage
+## Getting Started
 
-Add the extension to your Maven installation:
+1. Make sure you are using **Maven 4.0.0-rc-3 or later**. Mason does not work with Maven 3.x.
+
+2. Create the file `.mvn/extensions.xml` **in your project root directory** (next to your `pom.yaml` or `pom.json5`):
 
 ```xml
-<extension>
-  <groupId>eu.maveniverse.maven.mason</groupId>
-  <artifactId>mason</artifactId>
-  <version>0.2.0</version>
-</extension>
+<?xml version="1.0" encoding="UTF-8"?>
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.2.0">
+  <extension>
+    <groupId>eu.maveniverse.maven.mason</groupId>
+    <artifactId>mason</artifactId>
+    <version>0.3.0</version>
+  </extension>
+</extensions>
 ```
 
-Then you can use alternative formats for your POM files:
+3. Replace your `pom.xml` with a `pom.yaml`, `pom.json5`, `pom.conf`, or `pom.toml` file.
+
+4. Run Maven as usual: `mvn verify`
+
+> **Note:** The `.mvn/extensions.xml` file must be in the project root directory (the same directory as your POM file). For multi-module projects, it only needs to be in the root project — child modules will inherit the extension.
+
+## Format Examples
+
+The following examples show how to write a POM in each supported format:
 
 ### YAML Example (pom.yaml)
 

--- a/README.md
+++ b/README.md
@@ -30,7 +30,7 @@ Mason is a Maven extension that provides alternative model readers for Maven POM
 </extensions>
 ```
 
-3. Replace your `pom.xml` with a `pom.yaml`, `pom.json5`, `pom.conf`, or `pom.toml` file.
+3. Replace your `pom.xml` with a `pom.yaml`, `pom.json`, `pom.conf`, or `pom.toml` file.
 
 4. Run Maven as usual: `mvn verify`
 

